src/HotChocolate/Core/src/Types/Types/Relay/Extensions/RelayIdFieldExtensions.cs

```diff
@@ -257,6 +257,10 @@ private static IExtendedType RewriteType(ITypeInspector typeInspector, ITypeInfo
         {
             IIdSerializer serializer =
                 completionContext.Services.GetService<IIdSerializer>() ??
+                // TODO: Possible type system memory optimization
+                // would be to singleton this IdSerializer,
+                // e.g. coalesce with ?? (_idSerializer ??= new IdSerializer())
+                // where _idSerializer is static
                 new IdSerializer();
```

Comment:
```suggestion
            IIdSerializer serializer =
                completionContext.Services.GetRequiredService<IIdSerializer>();
```

Should always be one.